plutus-core/plutus-core/src/PlutusCore/Error.hs

```diff
@@ -103,15 +105,25 @@ data ParserErrorBundle
     deriving stock (Show, Eq, Generic)
     deriving anyclass (NFData)
 
-data Error uni fun ann
+data Error uni fun a b
     = ParseErrorE ParserErrorBundle
-    | UniqueCoherencyErrorE (UniqueError ann)
-    | TypeErrorE (TypeError (Term TyName Name uni fun ()) uni fun ann)
-    | NormCheckErrorE (NormCheckError TyName Name uni fun ann)
+    | UniqueCoherencyErrorE (UniqueError a)
+    | TypeErrorE (TypeError (Term TyName Name uni fun b) uni fun a)
```

Comment:
Do we actually need this? The typechecker always uses unit-annotated terms here. And if it didn't we'd probably want it to use the same annotation type as we're using everywhere else. So I would expect this to always be `()` or `a`. Do you have a usecase in mind?

Reply:
Yep, I need 
```haskell
compileToReadable
  :: (Compiling m e uni fun a, b ~ Provenance a)
  => Term TyName Name uni fun a
  -> m (Term TyName Name uni fun b)
```
to take a term with non-unit annotation (calling typechecker is one thing it does but it also does other simplification that needs the annotation), but this is not possible since in `Compiling`, `a` is hardcoded to `()`.

Reply:
Wait, I don't think this is correct; let me take a closer look

Reply:
Sure, but I don't think that's because of *this* unit.